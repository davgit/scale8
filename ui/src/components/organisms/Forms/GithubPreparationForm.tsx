import { FC } from 'react';
import { Box, DialogContent } from '@mui/material';
import DialogActionsWithCancel from '../../molecules/DialogActionsWithCancel';
import {
    GitHubPreparationFormProps,
    GitHubPreparationValues,
} from '../../../dialogPages/global/GithubAccount';
import FormFlex from '../../atoms/FormFlex';
import FormError from '../../atoms/FormError';
import { DialogFormContextProvider } from '../../../context/DialogFormContext';
import { DialogFormTextInput } from '../../atoms/DialogFormInputs/DialogFormTextInput';

const GithubPreparationForm: FC<GitHubPreparationFormProps> = (
    props: GitHubPreparationFormProps,
) => {
    return (
        <DialogFormContextProvider<GitHubPreparationValues> formProps={props}>
            <Box display="flex" flexDirection="column">
                {props.ssoError !== '' && (
                    <Box mb={2} width="100%">
                        <FormError error={props.ssoError} />
                    </Box>
                )}
                <FormFlex handleSubmit={props.handleSubmit}>
                    <DialogContent
                        sx={{
                            minHeight: '248px',
                        }}
                    >
                        <DialogFormTextInput name="githubUser" label="GitHub User" autoFocus />
                    </DialogContent>
                    <DialogActionsWithCancel
                        disableSubmit={props.isSubmitting}
                        handleDialogClose={props.handleDialogClose}
                        confirmText={props.submitText}
                    />
                </FormFlex>
            </Box>
        </DialogFormContextProvider>
    );
};

export default GithubPreparationForm;
