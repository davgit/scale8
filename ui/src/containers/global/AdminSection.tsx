import { FC } from 'react';
import { useQuery } from '@apollo/client';
import SignUpApprovalIcon from '../../components/atoms/Icons/SignUpApprovalIcon';
import AdminDashboardIcon from '../../components/atoms/Icons/AdminDashboardIcon';
import NavAdminQuery from '../../gql/queries/NavAdminQuery';
import { NavAdmin } from '../../gql/generated/NavAdmin';
import { buildAdminButtonProps, buildTabButtonProps } from '../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../abstractions/Section';
import { SectionKey } from '../SectionsDetails';
import { ChildrenOnlyProps } from '../../types/props/ChildrenOnlyProps';
import { toAdmin, toSignupApproval } from '../../utils/NavigationPaths';
import { useRouter } from 'next/router';
import { PageMenuButtonProps } from '../../components/molecules/SideMenuButton';

export const buildAdminTabsMenu = (): PageMenuButtonProps[] => [
    {
        icon: () => <AdminDashboardIcon />,
        label: 'Dashboard',
        link: toAdmin,
    },
    {
        icon: () => <SignUpApprovalIcon />,
        label: 'Signup Approval',
        link: toSignupApproval,
    },
];

const AdminSection: FC<ChildrenOnlyProps> = (props: ChildrenOnlyProps) => {
    const router = useRouter();

    const sectionProps: SectionProps<NavAdmin> = {
        // Admin = all permissions
        extractOrgUserDetails: () => ({
            id: '',
            name: '',
            is_paid: false,
            tag_manager_account: {
                id: '',
                trial_expires_in: 0,
                trial_expired: false,
                is_trial: false,
                enabled: true,
            },
            data_manager_account: {
                id: '',
                trial_expires_in: 0,
                trial_expired: false,
                is_trial: false,
                enabled: true,
            },
            me: {
                id: '',
                owner: false,
                permissions: {
                    can_view: true,
                    can_create: true,
                    can_edit: true,
                    can_delete: true,
                    is_admin: true,
                },
            },
        }),
        sectionKey: SectionKey.admin,
        queryResult: useQuery<NavAdmin>(NavAdminQuery),
        buildButtonsProps: (data) => [
            buildAdminButtonProps(router, data.me.orgs),
            buildTabButtonProps(router, buildAdminTabsMenu()),
        ],
        buildMenuItemsProps: () => buildAdminTabsMenu(),
        children: props.children,
    };

    return <Section<NavAdmin> {...sectionProps} />;
};

export default AdminSection;
