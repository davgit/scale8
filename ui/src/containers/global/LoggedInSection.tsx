import * as React from 'react';
import { FC, MouseEventHandler, useEffect, useReducer, useState } from 'react';
import LoggedInTemplate from '../../components/templates/containers/LoggedInTemplate';
import { ApolloError, useQuery } from '@apollo/client';
import { LoggedUser } from '../../gql/generated/LoggedUser';
import LoggedUserQuery from '../../gql/queries/LoggedUserQuery';
import { createHash } from 'crypto';
import { useCancelConfirmDialog } from '../../hooks/dialog/useCancelConfirmDialog';
import { CancelConfirmDialogProps } from '../../types/props/CancelConfirmDialogProps';
import { pageActions } from '../../actions/PageActions';
import { dialogReducer } from '../../context/DialogReducer';
import { orgUserReducer } from '../../context/OrgUserReducer';
import GithubCheck from '../../dialogPages/global/GithubCheck';
import { ChildrenOnlyProps } from '../../types/props/ChildrenOnlyProps';
import Navigate from '../../components/atoms/Next/Navigate';
import { dialogInit } from '../../context/LoggedInState';
import { useAppContext } from '../../context/AppContext';
import Loader from '../../components/organisms/Loader';
import { configStateFromData } from '../../context/ConfigState';
import { SectionKey } from '../SectionsDetails';
import { clearAuthSession } from '../../utils/authUtils';

export type UserSelectorProps = {
    loading: boolean;
    imgSrc?: string;
    userName: string;
    userEmail: string;
    handleAccountClick: MouseEventHandler;
    handleLogoutClick: MouseEventHandler;
};

export type SideBarProps = {
    handleNotificationClick: MouseEventHandler;
    children?: React.ReactNode;
    userSelectorProps: UserSelectorProps;
    isAdmin: boolean;
    unreadNotifications: number;
};

export type LoggedInProps = {
    sideBarProps: SideBarProps;
    children: React.ReactNode;
    breadcrumb: React.ReactNode;
    sideMenu: React.ReactNode;
    cancelConfirmDialogProps: CancelConfirmDialogProps;
};

const LoggedInSection: FC<ChildrenOnlyProps> = (props: ChildrenOnlyProps) => {
    const { setState, loggedInState, configState } = useAppContext();

    const [logOut, setLogOut] = useState(false);

    const [dialogState, dispatchDialogAction] = useReducer(dialogReducer, null, dialogInit);

    const [refreshCurrentPage, setRefreshCurrentPage] = useState(false);
    const [refreshCurrentSection, setRefreshCurrentSection] = useState(false);

    const [snackbarError, setSnackbarError] = useState<ApolloError | null>(null);

    const [orgUserState, dispatchOrgUserAction] = useReducer(orgUserReducer, null);

    const [section, setSection] = useState<symbol>(SectionKey.loggedOut);

    const [sectionHasAnalytics, setSectionHasAnalytics] = useState(false);

    const [gates, dispatchGatesAction] = useReducer(
        (
            state: Record<string, JSX.Element>,
            action: { type: 'set'; payload: { gateName: string; element: JSX.Element } },
        ): Record<string, JSX.Element> => {
            return { ...state, [action.payload.gateName]: action.payload.element };
        },
        {},
    );

    const teleport = (gateName: string, element: JSX.Element) => {
        dispatchGatesAction({ type: 'set', payload: { gateName, element } });
    };

    const { loading, error, data, ...queryResult } = useQuery<LoggedUser>(LoggedUserQuery);

    // Single Question Dialog
    const { dialogProps: cancelConfirmDialogProps, ask } = useCancelConfirmDialog();

    useEffect(() => {
        setState({
            loggedInState: {
                templateInteractions: {
                    ask,
                    dialogState,
                    dispatchDialogAction,
                    snackbarError,
                    setSnackbarError,
                    refreshCurrentPage,
                    setRefreshCurrentPage,
                    refreshCurrentSection,
                    setRefreshCurrentSection,
                    section,
                    setSection,
                    sectionHasAnalytics,
                    setSectionHasAnalytics,
                },
                loggedInUserState: {
                    loggedUserId: data?.me.id ?? '',
                    userIsAdmin: !!data?.me.is_admin,
                    invites: data ? data.me.invites.length : 0,
                    notifications: data ? data.me.user_notifications.length : 0,
                },
                orgUserState,
                dispatchOrgUserAction,
                gates,
                teleport,
            },
            configState: configStateFromData(data),
        });
    }, [
        data,
        dialogState,
        refreshCurrentPage,
        refreshCurrentSection,
        snackbarError,
        orgUserState,
        section,
    ]);

    if (error || logOut) {
        return <Navigate to="/login" />;
    }

    if (loading || loggedInState === undefined || configState === undefined) {
        return <Loader />;
    }

    const gravatarHash = createHash('md5')
        .update(data ? data.me.email.trim().toLowerCase() : '')
        .digest('hex');

    const userSelectorProps: UserSelectorProps = {
        imgSrc: `https://www.gravatar.com/avatar/${gravatarHash}?s=100&d=identicon`,
        loading,
        userName: `${data?.me.first_name} ${data?.me.last_name}`,
        userEmail: `${data?.me.email}`,
        handleAccountClick: () => {
            pageActions.manageAccount({
                dispatchDialogAction,
                refresh: () => {
                    (async () => {
                        await queryResult.refetch();
                    })();
                },
            });
        },
        handleLogoutClick: () => {
            clearAuthSession();
            setLogOut(true);
        },
    };

    const sideBarProps: SideBarProps = {
        handleNotificationClick: () => {
            pageActions.showNotification({
                dispatchDialogAction,
            });
        },
        userSelectorProps,
        isAdmin: data ? data.me.is_admin : false,
        unreadNotifications: data ? data.me.invites.length + data.me.user_notifications.length : 0,
    };

    return (
        <>
            <GithubCheck
                githubUser={data?.me.github_user ?? null}
                githubConnected={data?.me.github_connected ?? false}
            />
            <LoggedInTemplate
                sideBarProps={sideBarProps}
                breadcrumb={gates['breadcrumb'] ?? <></>}
                sideMenu={gates['side'] ?? <></>}
                cancelConfirmDialogProps={cancelConfirmDialogProps}
            >
                {props.children}
            </LoggedInTemplate>
        </>
    );
};

export default LoggedInSection;
