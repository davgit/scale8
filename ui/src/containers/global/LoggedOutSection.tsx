import { FC, useEffect } from 'react';
import LoggedOutTemplate from '../../components/templates/containers/LoggedOutTemplate';
import { ChildrenOnlyProps } from '../../types/props/ChildrenOnlyProps';
import { clearAuthSession } from '../../utils/authUtils';

const LoggedOutSection: FC<ChildrenOnlyProps & { stayLogged?: boolean }> = ({
    children,
    stayLogged,
}) => {
    useEffect(() => {
        if (!stayLogged) {
            clearAuthSession();
        }
    }, [stayLogged]);

    return <LoggedOutTemplate>{children}</LoggedOutTemplate>;
};

export default LoggedOutSection;
