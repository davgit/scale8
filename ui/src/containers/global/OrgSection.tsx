import { FC } from 'react';
import { useQuery } from '@apollo/client';
import NavOrgQuery from '../../gql/queries/NavOrgQuery';
import { NavOrg } from '../../gql/generated/NavOrg';
import {
    BreadcrumbButtonProps,
    buildOrgButtonProps,
    buildTabButtonProps,
} from '../../utils/BreadcrumbButtonsUtils';
import OrgDashboardIcon from '../../components/atoms/Icons/OrgDashboardIcon';
import OrgUsersIcon from '../../components/atoms/Icons/OrgUsersIcon';
import UserInviteIcon from '../../components/atoms/Icons/UserInviteIcon';
import { Section, SectionItem, SectionProps } from '../abstractions/Section';
import { SectionKey } from '../SectionsDetails';
import OrgSettingsIcon from '../../components/atoms/Icons/OrgSettingsIcon';
import { useConfigState } from '../../context/AppContext';
import { NextRouter, useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../types/props/ChildrenAndIdProps';
import { toOrg } from '../../utils/NavigationPaths';
import { PageMenuButtonProps } from '../../components/molecules/SideMenuButton';
import { CurrentOrgPermissions } from '../../context/OrgUserReducer';
import { SvgIconProps } from '@mui/material';

export const buildOrgTabsMenu = (
    orgPermissions: CurrentOrgPermissions,
    id: string,
    useSignup: boolean,
): PageMenuButtonProps[] => {
    return [
        ...[
            {
                icon: () => <OrgDashboardIcon />,
                label: 'Services',
                link: toOrg({ id }, 'dashboard'),
            },
            {
                icon: () => <OrgUsersIcon />,
                label: 'Users',
                link: toOrg({ id }, 'users'),
                disabled: !orgPermissions.isAdmin,
            },
        ],
        ...(useSignup
            ? [
                  {
                      icon: () => <UserInviteIcon />,
                      label: 'User Invites',
                      link: toOrg({ id }, 'invites'),
                      disabled: !orgPermissions.isAdmin,
                  },
              ]
            : []),
        {
            icon: () => <OrgSettingsIcon />,
            label: 'Settings',
            link: toOrg({ id }, 'settings'),
            disabled: !orgPermissions.isAdmin,
        },
    ];
};

export const buildOrgButtons = (
    orgs: SectionItem[],
    currentOrg: SectionItem,
    router: NextRouter,
    orgPermissions: CurrentOrgPermissions,
    useSignup: boolean,
    forceCurrentEntry?: string,
    forceCurrentIcon?: FC<SvgIconProps>,
    isCurrentPage = false,
): BreadcrumbButtonProps[] => [
    buildOrgButtonProps(router, orgs, currentOrg.id, currentOrg.name, isCurrentPage),
    buildTabButtonProps(
        router,
        buildOrgTabsMenu(orgPermissions, currentOrg.id, useSignup),
        forceCurrentEntry,
        forceCurrentIcon,
    ),
];

const OrgSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { useSignup } = useConfigState();

    const sectionProps: SectionProps<NavOrg> = {
        children,
        sectionKey: SectionKey.org,
        extractOrgUserDetails: (data) => data.getOrg,
        queryResult: useQuery<NavOrg>(NavOrgQuery, {
            variables: { id },
        }),
        buildButtonsProps: (data, orgPermissions) => {
            return buildOrgButtons(
                data.me.orgs,
                data.getOrg,
                router,
                orgPermissions,
                useSignup,
                undefined,
                undefined,
                true,
            );
        },
        buildMenuItemsProps: (_, orgPermissions) => {
            return buildOrgTabsMenu(orgPermissions, id, useSignup);
        },
    };

    return <Section<NavOrg> {...sectionProps} />;
};

export default OrgSection;
