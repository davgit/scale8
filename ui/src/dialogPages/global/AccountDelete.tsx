import { FC } from 'react';
import { DialogPageProps } from '../../types/DialogTypes';
import { Box, Button, DialogContent, lighten } from '@mui/material';
import DialogActionsWithCancel from '../../components/molecules/DialogActionsWithCancel';
import { useMutation } from '@apollo/client';
import Loader from '../../components/organisms/Loader';
import { DeleteAccount } from '../../gql/generated/DeleteAccount';
import DeleteAccountQuery from '../../gql/mutations/DeleteAccountQuery';
import FormGqlError from '../../components/atoms/FormGqlError';
import { useLoggedInState } from '../../context/AppContext';
import Navigate from '../../components/atoms/Next/Navigate';
import { toLogin } from '../../utils/NavigationPaths';
import { clearAuthSession } from '../../utils/authUtils';

const AccountDelete: FC<DialogPageProps> = (props: DialogPageProps) => {
    const { templateInteractions } = useLoggedInState();
    const { ask } = templateInteractions;
    const { handleDialogClose } = props;

    const [deleteAccount, { loading, data, error: gqlError }] =
        useMutation<DeleteAccount>(DeleteAccountQuery);

    const successfullySubmitted = data?.deleteMe;

    if (successfullySubmitted) {
        clearAuthSession();
        handleDialogClose(false);
        return <Navigate to={toLogin} />;
    }

    if (loading) {
        return <Loader />;
    }

    return (
        <Box display="flex" flexDirection="column">
            <DialogContent
                sx={{
                    display: 'flex',
                    flexDirection: 'column',
                    minHeight: '248px',
                }}
            >
                <FormGqlError error={gqlError} />
                <small>
                    If you delete your account you will be a automatically removed from all
                    organizations.
                    <br />
                    <br />
                    If you are the owner of an organization you cannot delete your account. you must
                    first transfer the ownership or remove the organization.
                    <br />
                    <br />
                </small>
                <Button
                    variant="contained"
                    onClick={() => {
                        ask(`Are you sure you want to delete your account?`, () => {
                            (async () => {
                                await deleteAccount();
                            })();
                        });
                    }}
                    sx={{
                        color: '#ffffff',
                        backgroundColor: (theme) => lighten(theme.palette.error.main, 0.4),
                        '&:hover': {
                            backgroundColor: (theme) => theme.palette.error.main,
                        },
                    }}
                    disableElevation
                >
                    Delete Account
                </Button>
            </DialogContent>
            <DialogActionsWithCancel handleDialogClose={props.handleDialogClose} />
        </Box>
    );
};

export { AccountDelete };
