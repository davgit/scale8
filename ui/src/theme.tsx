import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
