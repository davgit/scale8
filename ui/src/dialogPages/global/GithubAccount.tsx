import { FC } from 'react';
import Loader from '../../components/organisms/Loader';
import { FormProps, useFormValidation } from '../../hooks/form/useFormValidation';
import { useMutation, useQuery } from '@apollo/client';
import LoggedUserQuery from '../../gql/queries/LoggedUserQuery';
import { LoggedUser } from '../../gql/generated/LoggedUser';
import GithubPreparationQuery from '../../gql/mutations/GithubPreparationQuery';
import GithubPreparationForm from '../../components/organisms/Forms/GithubPreparationForm';
import { PrepareGitHubLinkInput } from '../../gql/generated/globalTypes';
import { GithubPreparationValues } from '../../gql/generated/GithubPreparationValues';
import nameValidator from '../../utils/validators/nameValidator';
import { DialogPageProps } from '../../types/DialogTypes';
import { queryLoaderAndError } from '../../abstractions/QueryLoaderAndError';
import { Box, DialogActions, DialogContent } from '@mui/material';
import FormGqlError from '../../components/atoms/FormGqlError';
import GithubAccountRemoveQuery from '../../gql/mutations/GithubAccountRemoveQuery';
import { GithubAccountRemoveValues } from '../../gql/generated/GithubAccountRemoveValues';
import GithubLoginForUser from './GithubLoginForUser';
import { logError } from '../../utils/logUtils';
import { DialogConfirmButton } from '../../components/atoms/DialogConfirmButton';
import { DialogCancelButton } from '../../components/atoms/DialogCancelButton';

type GithubAccountAfterLoadProps = DialogPageProps & {
    hasLinkedSSO: boolean;
    githubUser: string | null;
};

export type GitHubPreparationValues = {
    githubUser: string;
};

export type GitHubPreparationFormProps = FormProps<GitHubPreparationValues> & {
    handleDialogClose: (checkChanges: boolean) => void;
};

const GithubAccountUseForm: FC<GithubAccountAfterLoadProps> = (
    props: GithubAccountAfterLoadProps,
) => {
    const [GithubPreparation, { loading, data, error: gqlError }] =
        useMutation<GithubPreparationValues>(GithubPreparationQuery);

    const submitForm = async (gitHubPreparationValues: GitHubPreparationValues) => {
        const prepareGitHubLinkInput: PrepareGitHubLinkInput = {
            github_user: gitHubPreparationValues.githubUser,
        };
        try {
            await GithubPreparation({
                variables: { prepareGitHubLinkInput },
            });
        } catch (error) {
            logError(error);
        }
    };

    const initialState = {
        githubUser: '',
    };

    const formValidationValues = useFormValidation<GitHubPreparationValues>(
        initialState,
        [
            {
                field: 'githubUser',
                validator: nameValidator,
                error: () => 'GitHub username too short',
            },
        ],
        submitForm,
    );

    if (loading) {
        return <Loader />;
    }

    if (data?.prepareGitHubLink.id !== undefined) {
        return <GithubLoginForUser {...props} />;
    }

    const formProps: GitHubPreparationFormProps = {
        ...formValidationValues,
        gqlError,
        submitText: 'Connect Github Account',
        title: 'Connect Github Account',
        handleDialogClose: props.handleDialogClose,
    };

    return <GithubPreparationForm {...formProps} />;
};

const RemoveGithubAccountUseForm: FC<GithubAccountAfterLoadProps> = (
    props: GithubAccountAfterLoadProps,
) => {
    const [GithubAccountRemove, { loading, data, error: gqlError }] =
        useMutation<GithubAccountRemoveValues>(GithubAccountRemoveQuery);

    if (loading) {
        return <Loader />;
    }

    if (data?.removeGitHubLink) {
        props.handleDialogClose(false);
    }

    return (
        <Box display="flex" flexDirection="column">
            <DialogContent sx={{ minHeight: 248 }}>
                <FormGqlError error={gqlError} />
                This account is connected to the github account:
                <br />
                <b>{props.githubUser}</b>
            </DialogContent>
            <DialogActions sx={{ padding: 2, justifyContent: 'center' }}>
                <DialogCancelButton onClick={() => props.handleDialogClose(false)}>
                    Cancel
                </DialogCancelButton>
                <DialogConfirmButton
                    onClick={() => {
                        (async () => {
                            try {
                                await GithubAccountRemove();
                            } catch (error) {
                                logError(error);
                            }
                        })();
                    }}
                >
                    Disconnect Github Account
                </DialogConfirmButton>
            </DialogActions>
        </Box>
    );
};

const GithubAccountAfterLoad: FC<GithubAccountAfterLoadProps> = (
    props: GithubAccountAfterLoadProps,
) => {
    if (props.hasLinkedSSO) {
        return <RemoveGithubAccountUseForm {...props} />;
    } else {
        return <GithubAccountUseForm {...props} />;
    }
};

const GithubAccount: FC<DialogPageProps> = (props: DialogPageProps) => {
    return queryLoaderAndError<LoggedUser>(
        false,
        useQuery<LoggedUser>(LoggedUserQuery),
        (data: LoggedUser) => {
            return (
                <GithubAccountAfterLoad
                    hasLinkedSSO={data.me.github_connected}
                    githubUser={data.me.github_user}
                    {...props}
                />
            );
        },
    );
};

export { GithubAccount };
