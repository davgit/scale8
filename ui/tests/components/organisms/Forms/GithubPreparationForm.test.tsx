import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import GithubPreparationForm from '../../../../src/components/organisms/Forms/GithubPreparationForm';
import { GitHubPreparationValues } from '../../../../src/dialogPages/global/GithubAccount';

const MockForm: FC = () => {
    const formProps = useMockFormProps<GitHubPreparationValues>({
        githubUser: '',
    });

    return <GithubPreparationForm {...formProps} ssoError="" />;
};

describe('GithubPreparationForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
