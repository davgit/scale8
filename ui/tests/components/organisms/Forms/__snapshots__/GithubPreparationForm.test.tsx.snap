// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubPreparationForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-j7qwjs"
  >
    <form
      class="MuiBox-root css-mjj15b"
    >
      <div
        class="MuiDialogContent-root css-1pn2u1k-MuiDialogContent-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
          >
            GitHub User
            <span
              aria-hidden="true"
              class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
            >
              â€‰*
            </span>
          </label>
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-1ejkwvn-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              name="githubUser"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-snbeu5-MuiDialogActions-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-7kx4b6-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Cancel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
